File: tests/python/test_suite.py
import os
import json

from fsoft_diff_patch import diff, patch

# for all directories in ../data directory, run the test
data_dir = os.path.join(os.path.dirname(__file__), '../data')
dirs = sorted(os.listdir(data_dir))

for dir in dirs:
    # write dir name, but keep the cursor at the same line
    print(f"Testing: {dir} ... ", end="")

    file01 = os.path.join(data_dir, dir, '01.json')
    file02 = os.path.join(data_dir, dir, '02.json')

    # read file 01.json and 02.json from the directory
    with open(file01, 'r') as f:
        obj01 = json.load(f)
    with open(file02, 'r') as f:
        obj02 = json.load(f)

    # diff obj01 and obj02 and create a diffData object
    diff_data = diff(obj01, obj02)

    # apply diffData to obj01 and create a new object
    patched_obj01 = patch(obj01, diff_data)

    # check if patched_obj01 is equal to obj02
    if json.dumps(patched_obj01) != json.dumps(obj02):
        print("FAILED")
    else:
        print("OK")

File: tests/dart/test_suite.dart
import 'dart:io';
import 'dart:convert';

import '../../dart/lib/fsoft_diff_patch.dart';

void main() {
  // for all directories in ../data directory, run the test
  var dataDir =
      Directory('../data');
  var dirs = dataDir
      .listSync()
      .where((e) => e is Directory)
      .map((e) => e.path)
      .toList()
      ..sort();

  for (final dir in dirs) {

    final dirName = dir.split("/").last;

    // write dir name, but keep the cursor at the same line
    stdout.write('Testing: $dirName ... ');

    final file01 = File('$dir/01.json');
    final file02 = File('$dir/02.json');

    // read file 01.json and 02.json from the directory
    final obj01 = jsonDecode(file01.readAsStringSync());
    final obj02 = jsonDecode(file02.readAsStringSync());

    // diff obj01 and obj02 and create a diffData object
    final diffData = diff(obj01, obj02);

    // apply diffData to obj01 and create a new object
    final patchedObj01 = patch(obj01, diffData);

    // check if patchedObj01 is equal to obj02
    if (jsonEncode(patchedObj01) != jsonEncode(obj02)) {
      print('FAILED');
    } else {
      print('OK');
    }
  }
  ;
}
